### DISCUSSION TITLE
Logical / Boolean Aggregation Functions

### DISCUSSION THREAD

**Discussion by gramian** [ArcadeData/arcadedb#854] [Ideas]
[PostgreSQL](https://www.postgresql.org/docs/9.5/functions-aggregate.html#FUNCTIONS-AGGREGATE-TABLE) and [data.world](https://docs.data.world/documentation/sql/reference/boolean_aggregations.html)  (and others) have some very interesting boolean aggregation functions, [BOOL_AND](https://docs.data.world/documentation/sql/reference/aggregations/bool_and.html) and [BOOL_OR](https://docs.data.world/documentation/sql/reference/aggregations/bool_or.html), which aggregate columns (here properties) based on a boolean expression (optionally into groups). While one can probably emulate these function with nested queries and `count`s, the appeal of these function would be their performant short-circuit implementation, meaning `AND` returns `false` as soon as the first `false` (in a group) is found, and `OR` returns `true` as soon as the first `true` (in a group) is found.

I would like to propose adding these two boolean aggregators, as they  should be really useful for data cleaning or data exploration, answering questions like "Are all of ..." or "Are any of ..." fast. Thus, they also fit well with  the _Play With Data_ motto. Lastly, these functions are different to filtering with [`all()` and `any()`](https://docs.arcadedb.com/#Filtering) due to different scope (property vs document) and are (only in purpose) similar to SQL's [`ALL` and `ANY`](https://www.w3schools.com/sql/sql_any_all.asp).

Thank you for considering my proposal

**Comment by lvca** [comment]
+1
